### PR TITLE
fix: restore masternode keys issues

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Refactor wallet creation for Onboarding
## Related PR's and Dependencies

None
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -78,7 +78,7 @@ platform :android do
   lane :test do |options|
     flavor=options[:flavor].capitalize
 	type=options[:type].capitalize
-    gradle(task: "test#{flavor}#{type}UnitTest")
+    gradle(task: "test#{flavor}#{type}UnitTest --info")
```
Comment (HashEngineering) [inline fastlane/Fastfile:81]: This is to find out why some tests are failing.  This must removed before review or merge.

File: wallet/src/de/schildbach/wallet/util/WalletUtils.java [hunk 2]
```diff
@@ -117,88 +117,6 @@ public static Editable formatHash(@Nullable final String prefix, final String ad
         return builder;
     }
 
-    public static Wallet restoreWalletFromProtobufOrBase58(final InputStream is,
-                                                           final NetworkParameters expectedNetworkParameters) throws IOException {
-        return restoreWalletFromProtobufOrBase58(is, expectedNetworkParameters, null);
-    }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/util/WalletUtils.java:123]: All restore wallet code was moved to WalletFactory.

File: wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        val context = Context(MainNetParams.get())
+        walletFactory.create(context.params)
+    }
+
+    @Test
+    fun restoreFromSeedTest() {
+        val walletFactory = DashWalletFactory(application)
+        val contextMocked = mockk<Context>()
+        every { contextMocked.params } returns MainNetParams.get()
+        every { Context.getOrCreate(any()) } returns contextMocked
```
[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt:55]: All tests should probably be mocked in similar fashion.

File: wallet/src/de/schildbach/wallet/service/WalletFactory.kt [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+    // Onboarding
+    fun create(params: NetworkParameters): Wallet
+    fun restoreFromSeed(params: NetworkParameters, recoveryPhrase: List<String>): Wallet
+    @Throws(IOException::class)
+    fun restoreFromFile(params: NetworkParameters, backupUri: Uri, password: String): Pair<Wallet, Boolean>
+
+    // loading from persistent storage
+    fun getExtensions(params: NetworkParameters): Array<WalletExtension>
+    fun load(params: NetworkParameters, walletFile: File): Wallet
+    fun restoreFromBackup(params: NetworkParameters, backupFile: String): Wallet
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/service/WalletFactory.kt:65]: These two methods are not used yet, but could replace what is in WalletApplication.  

In a future state we may consider a `WalletManager` which will implement `WalletDataProvider` and move wallet related code out of WalletApplication with respect to setting the wallet object after onboarding and loading the wallet from persistent storage.

File: wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Syn-McJ) [reply wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt:68]: Fastlane might be running tests in a testnet environment, so the testnet context might be propagated somewhere before this.

**Review by Syn-McJ — APPROVED** [review]
Great work

File: wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (HashEngineering) [reply wallet/test/de/schildbach/wallet/util/services/WalletFactoryTest.kt:68]: We have this problem in DashJ.  One interesting about BitcoinJ 0.17 (we are still on 0.15) is that they removed the NetworkParameters field from Context which solves this problem.

**Pull request merged**